### PR TITLE
Add support for repo setting `use_squash_pr_title_as_default`

### PR DESCRIPTION
**Pull request by garrettheel**
Adds support for the repository setting `use_squash_pr_title_as_default`.

Related API docs:
https://docs.github.com/en/rest/repos/repos#get-a-repository
https://docs.github.com/en/rest/repos/repos#create-an-organization-repository
https://docs.github.com/en/rest/repos/repos#update-a-repository

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -26,52 +26,53 @@ type RepositoriesService service
...
+	Organization              *Organization   `json:"organization,omitempty"`
+	Permissions               map[string]bool `json:"permissions,omitempty"`
+	AllowRebaseMerge          *bool           `json:"allow_rebase_merge,omitempty"`
+	AllowUpdateBranch         *bool           `json:"allow_update_branch,omitempty"`
+	AllowSquashMerge          *bool           `json:"allow_squash_merge,omitempty"`
+	AllowMergeCommit          *bool           `json:"allow_merge_commit,omitempty"`
+	AllowAutoMerge            *bool           `json:"allow_auto_merge,omitempty"`
+	AllowForking              *bool           `json:"allow_forking,omitempty"`
+	DeleteBranchOnMerge       *bool           `json:"delete_branch_on_merge,omitempty"`
+	UseSquashPrTitleAsDefault *bool           `json:"use_squash_pr_title_as_default"`
```
Comment (gmlewis) [inline github/repos.go:72]:
```suggestion
	UseSquashPRTitleAsDefault *bool           `json:"use_squash_pr_title_as_default,omitempty"`
```

File: github/repos.go [hunk 2]
```diff
@@ -362,16 +363,17 @@ type createRepoRequest struct {
...
+	GitignoreTemplate         *string `json:"gitignore_template,omitempty"`
+	LicenseTemplate           *string `json:"license_template,omitempty"`
+	AllowSquashMerge          *bool   `json:"allow_squash_merge,omitempty"`
+	AllowMergeCommit          *bool   `json:"allow_merge_commit,omitempty"`
+	AllowRebaseMerge          *bool   `json:"allow_rebase_merge,omitempty"`
+	AllowUpdateBranch         *bool   `json:"allow_update_branch,omitempty"`
+	AllowAutoMerge            *bool   `json:"allow_auto_merge,omitempty"`
+	AllowForking              *bool   `json:"allow_forking,omitempty"`
+	DeleteBranchOnMerge       *bool   `json:"delete_branch_on_merge,omitempty"`
+	UseSquashPrTitleAsDefault *bool   `json:"use_squash_pr_title_as_default"`
```
Comment (gmlewis) [inline github/repos.go:376]:
```suggestion
	UseSquashPRTitleAsDefault *bool   `json:"use_squash_pr_title_as_default,omitempty"`
```

File: github/repos.go [hunk 3]
```diff
@@ -397,26 +399,27 @@ func (s *RepositoriesService) Create(ctx context.Context, org string, repo *Repo
...
+		GitignoreTemplate:         repo.GitignoreTemplate,
+		LicenseTemplate:           repo.LicenseTemplate,
+		AllowSquashMerge:          repo.AllowSquashMerge,
+		AllowMergeCommit:          repo.AllowMergeCommit,
+		AllowRebaseMerge:          repo.AllowRebaseMerge,
+		AllowUpdateBranch:         repo.AllowUpdateBranch,
+		AllowAutoMerge:            repo.AllowAutoMerge,
+		AllowForking:              repo.AllowForking,
+		DeleteBranchOnMerge:       repo.DeleteBranchOnMerge,
+		UseSquashPrTitleAsDefault: repo.UseSquashPrTitleAsDefault,
```
Comment (gmlewis) [inline github/repos.go:422]:
```suggestion
		UseSquashPRTitleAsDefault: repo.UseSquashPRTitleAsDefault,
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @garrettheel !
Just a few minor tweaks, please, before we merge.

**Comment by garrettheel** [comment]
@gmlewis thanks for reviewing, addressed your comments!

**Review by gmlewis — APPROVED** [review]
Thank you, @garrettheel !
LGTM.
Merging.

**Pull request merged**